sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2943,3 +2943,87 @@ case class Sentences(
     copy(str = newFirst, language = newSecond, country = newThird)
 
 }
+
+/**
+ * Splits a given string by a specified delimiter.
```

Comment:
This doesn't explain the difference between this expression and the `StringSplit`, can we say more about the major difference? We can also name it `StringSplitSQL` to match `UTF8String.splitSQL`